### PR TITLE
[LANG-1593] - Common behaviour for StringUtils join APIs when called with char or String delimiter

### PR DESCRIPTION
**Pull request by HubertWo**
JIRA:  https://issues.apache.org/jira/browse/LANG-1593

### CODE REVIEW

File: src/main/java/org/apache/commons/lang3/StringUtils.java [hunk 1]
```diff
@@ -3900,6 +3900,41 @@ public static String join(final byte[] array, final char separator) {
...
+     * @param separator
+     *            the separator String to use
+     * @param startIndex
+     *            the first index to start joining from. It is an error to pass in a start index past the end of the
+     *            array
+     * @param endIndex
+     *            the index to stop joining from (exclusive). It is an error to pass in an end index past the end of
+     *            the array
+     * @return the joined String, {@code null} if null array input
+     * @since 3.2
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/lang3/StringUtils.java:3935]: You're not adding method to the old version 3.2; the next version will be 3.12.

Comment (HubertWo) [reply src/main/java/org/apache/commons/lang3/StringUtils.java]: My mistake. Fixed.

**Comment by arturobernalg** [comment]
LGTM

**Comment by HubertWo** [comment]
@garydgregory May I ask you for second review and approval? :)

**Comment by HubertWo** [comment]
Pulled changes from apache:master.

**Comment by HubertWo** [comment]
Pulled changes from apache:master for the third time.

**Pull request closed without merging**